### ISSUE TITLE
Memory Leak of  cgo

### ISSUE THREAD

**Issue by boobusy** [vulkan-go/vulkan#72]
![image](https://user-images.githubusercontent.com/4724750/218548586-6e04e983-d114-4150-9efd-8768a0893800.png)

The memory viewing with pprof is very small.

**Comment by xlab** [comment]
@boobusy is it a false alarm?

**Comment by boobusy** [comment]
> 
@xlab 
 The use of PresentMode = vk.PresentModeImmediate  Memory growth is fast.

**Comment by boobusy** [comment]
maybe try "github.com/cannibalvox/cgoparam" Memory allocator.

**Comment by xlab** [comment]
@boobusy good idea, although I think this would beg for a full rewrite for my cgo generator xD

**Comment by akyoto** [comment]
I stumbled upon this on my Linux machine as well, with `PresentMode = vk.PresentModeImmediate` you can observe the memory growing very quickly.
